tsdb/engine/tsm1/digest_reader.go

```diff
@@ -28,10 +29,22 @@ func (r *DigestReader) ReadManifest() (*DigestManifest, error) {
 	lr := io.LimitReader(r.sr, int64(n))
 
 	m := &DigestManifest{}
-	return m, json.NewDecoder(lr).Decode(m)
+	if err := json.NewDecoder(lr).Decode(m); err != nil {
```

Comment:
given this is an exported method, you may want to add a guard at the entry point that returns an error if the manifest has already been read and this API is called again.

```go
if r.manifestRead {
    return nil, <some error>
}
```

Reply:
Good point. What do you thing about caching the manifest and returning it on future reads? Would that make sense or should that be up to the caller?

Reply:
Sounds fair to me, if you think that is a valid use case

Reply:
After thinking about it, returning an error probably makes more sense with a method named `Read...` and it's easy enough for the caller to store it if needed.